src/collections/componentSet.ts

```diff
@@ -292,6 +295,20 @@ export class ComponentSet extends LazyCollection<MetadataComponent> {
 
     const mdapiDeploy = new MetadataApiDeploy(operationOptions);
     await mdapiDeploy.start();
+
+    // Creates an array of unique metadata types to be deployed, uses Set to avoid duplicates.
+    const listOfMetadataTypesDeployed = Array.from(new Set(toDeploy.map((c) => c.type.name)));
+
+    void Lifecycle.getInstance().emitTelemetry({
+      eventName: 'metadata_api_deploy',
+      library: 'SDR',
+      function: 'ComponentSet.deploy',
+      totalNumberOfComponents: this.size,
+      numberOfComponentsDeployed: toDeploy.length,
+      componentsDeployed: listOfMetadataTypesDeployed.toString(),
+      componentsDeployedTruncated: listOfMetadataTypesDeployed.length < 8000 ? false : true,
```

Comment:
Just curious about the number 8000. Is this a limit known to the customer?

Reply:
That number is for unique metadata types in the deploy request.  Salesforce has around 350 metadata types so not sure where 8000 comes from.  Also, what's being truncated?

Reply:
@shetzel not entire sure, I just copied this from the PR where this telemetry was removed from toolbelt.
https://github.com/salesforcecli/toolbelt/pull/215/files#diff-ca446f5ccf231e22c85434264adcb8d977f76d6375c1ae789fd39c1bde5aa324L330